data/lua/explode_ability.lua

```diff
@@ -0,0 +1,38 @@
+
+local _ = wesnoth.textdomain 'wesnoth-help'
+local T = wml.tag
+
+-- The explode event code
+wesnoth.game_events.add_repeating("die", function()
+	local ec = wesnoth.current.event_context
+
+	if not ec.x1 or not ec.y1 then
+		return
+	end
+
+	local u_exploder = wesnoth.units.get(ec.x1, ec.y1)
+
+	if not u_exploder or not u_exploder:matches { ability = "explode" } then
+		return
+	end
+
+	local u_exploder_cfg = u_exploder.__cfg
+        local adjacent_units = wesnoth.units.find_on_map{T.filter_adjacent { id = u_exploder_cfg.id}}
+
+	for i,v in ipairs(wml.get_child(u_exploder_cfg, "abilities")) do
+		if v[1] == "dummy" and v[2].id == "explode" then
```

Comment:
It looks like you forgot to update this check. Both strings should read "explode".

Also, referring to the tag name as `v[1]` and its content as `v[2]` should be considered deprecated at least for mainline code. Prefer referring to `v.tag` and `v.contents` instead.